src/js/controller/controller.js

```diff
@@ -234,9 +234,11 @@ Object.assign(Controller.prototype, {
             eventsReadyQueue = null;
 
             _model.change('viewable', viewableChange);
-            _model.change('viewable', _checkPlayOnViewable);
-            _model.once('change:autostartFailed change:autostartMuted change:mute', function(model) {
-                model.off('change:viewable', _checkPlayOnViewable);
+            _model.once('change:canAutoplay', function(model) {
+                model.once('change:autostartFailed change:autostartMuted change:mute', function() {
```

Comment:
"autostartMuted" is set as a result of "canAutoplay" which could remove the "change:viewable" listener before it had a chance to work.

Reply:
I was thinking to refactor the `canAutoplay`, `autostart`, and `autostartMuted` to `_checkAutostart`, as well as these handlers there. This will make it easier to emit `autostartNotAllowed` in the different scenarios, tidy things up, and leave `_autoStart` untouched.

Reply:
See https://github.com/jwplayer/jwplayer/pull/2743/files